helix-core/src/main/java/org/apache/helix/controller/stages/MessageGenerationPhase.java

```diff
@@ -190,7 +192,8 @@ public void process(ClusterEvent event) throws Exception {
 
   private Message createMessage(HelixManager manager, String resourceName, String partitionName,
       String instanceName, String currentState, String nextState, String sessionId,
-      String stateModelDefName, String stateModelFactoryName, int bucketSize) {
```

Comment:
This method has too many parameters, we need to just pass in resource or have a message builder class